subworkflows/local/bam_variant_calling_mpileup/main.nf

```diff
@@ -6,71 +6,62 @@ include { GATK4_MERGEVCFS                } from '../../../modules/nf-core/gatk4/
 
 workflow BAM_VARIANT_CALLING_MPILEUP {
     take:
-        cram                    // channel: [mandatory] [meta, cram, interval]
-        fasta                   // channel: [mandatory]
-        dict
+    cram      // channel: [mandatory] [ meta, cram, crai ]
+    dict      // channel: [mandatory] [ meta, dict ]
+    fasta     // channel: [mandatory] [ fasta ]
+    intervals // channel: [mandatory] [ intervals, num_intervals ] or [ [], 0 ] if no intervals
 
     main:
+    versions = Channel.empty()
 
-    ch_versions = Channel.empty()
+    // Combine cram and intervals for spread and gather strategy
+    cram_intervals = cram.combine(intervals)
+        // Move num_intervals to meta map and reorganize channel for BCFTOOLS_MPILEUP/SAMTOOLS_MPILEUP modules
+        .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, intervals ] }
 
     keep_bcftools_mpileup = false
```

Comment:
Could this option be added to the input of the subwf?

Reply:
which option?

Reply:
`keep_bcftools_mpileup` (damn extra lines when selecting one line :p)

Reply:
We'll definitively think about it